src/pages/Vault/index.tsx

```diff
@@ -210,6 +217,8 @@ const Vault = (): JSX.Element => {
     }
   ];
 
+  const interactionEnabled = vaultClientLoaded && address === selectedVaultAddress;
```

Comment:
For data which is being called from inside of a component (and which we can expect to change over time) it's better to use state because otherwise you'll be reinitialising the variable when you don't need to.

To take address as an example - this is being pulled into the component from general state, and will update multiple times (not only when a user changes account, but also because of the underlying design of the application).

If you set `interactionEnabled` using state and update it with `useEffect` this will only be changed when either the `address` or the `vaultClientLoaded` status changes. Setting it as a plain variable causes it to be reinitialised on every re-render, even when the relevant dependencies (address, vaultClientLoaded) haven't changed.

There are use cases for using a plain variable - If you were passing only the `address` and `vaultCientLoaded` props into simple component and running a single function against them, there would be no benefit to using state as a re-render would only be triggered whenever one of those values change.

In this case though there are multiple things which can trigger a re-render (`bridgeLoaded`, `capacity`, `totalIssueRequests` etc.). A plain variable will be reinitialised every single time one of those things updates and triggers a re-render. A value held in state and being updated using `useEffect` will only update whenever one of the values in the dependency array changes.

Take this code as an example (from the transfer feature as I don't have a vault running locally).

```
  const [addressState, setAddressState] = React.useState<string | undefined>(undefined);

  React.useEffect(() => {
    if (!addressState) {
      setAddressState(address);
    }

    console.log('addressState', addressState);
  }, [address, addressState]);

  const addressVariable = address;

  console.log('addressVariable', addressVariable);
```

Both of those values are being set pulling an address from `state.general` in the same way as you're doing in the vault. This is the log from loading that page and then entering values into the fields etc.:

<img width="762" alt="Screenshot 2022-01-27 at 16 24 24" src="https://user-images.githubusercontent.com/40243778/151400442-6b012d66-4e07-4162-a404-0e3c78b4fb04.png">

You can see that `addressState` is logged out twice - initially because it's undefined, and then again when the address is set. You can see `addressVariable` being logged out multiple times, initially as the page re-renders multiple times when loading, and then again when I change values in the form field (those 10 logs at the bottom).

Reply:
That screenshot is missing the first render of addressState, but you can see it here:

<img width="510" alt="Screenshot 2022-01-27 at 16 31 54" src="https://user-images.githubusercontent.com/40243778/151402030-9f2f2e08-6be3-4db8-9a45-d0c4b9ae2612.png">

Reply:
I know that technically all that demonstrates is that the console log is running on every page render :) but hopefully it's clear that everything that is executed directly in the function body of a component happens every time the component re-renders.

Reply:
Hi Anton, you're overcomplicating this and this is a change that needs to be made. The way React handles this is very simple. Unless you want a variable reinitialised on every render (which we don't, because as described above it will often remain the same between renders) it needs to be either set in state or memoized. It's only worth memoizing if it's an expensive calculation which it's not, so it needs to be handled in state.

Please make the change as requested and then I can approve the PR.

Reply:
OK, I'll make that change because I want this PR to be merged.
But please send me an article explaining that theory as I requested.